### PR TITLE
Sidebar service

### PR DESCRIPTION
**Pull request by crystallow1168**
Hi Michael! Could you please review my PR? Thanks!

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -8,9 +8,9 @@ module.exports = {
 
   moduleFileExtensions: ['js', 'json', 'jsx'],
 
-  setupFiles: ['<rootDir>/test/enzyme.config.js'],
+  setupFiles: ['<rootDir>/tests/enzyme.config.js'],
 
-  testEnvironment: 'jsdom',
+  testEnvironment: 'node',
```
Comment (miicow) [inline jest.config.js:13]: How come you changed the testEnvironment from jsdom to node?

File: client/src/components/App.jsx [hunk 2]
```diff
@@ -1,14 +1,86 @@
...
 
+    this.state = {
+    };
+  }
+
+  componentDidMount() {
+    $.ajax({
+      url: 'http://localhost:3333/r',
+      success: (results) => {
+        const restaurantInfo = results.map((each) => {
```
Comment (miicow) [inline client/src/components/App.jsx]: Is each object? if yes, maybe its possible for deconstructing here?

File: tests/server.test.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+const app = require('../server/index');
+
+describe('GET /r', function () {
+  it ('responds with json', function(done) {
+    request(app)
+    .get('/r')
+    .expect('Content-Type', /json/)
+    .expect(200, done)
+  });
+})
```
Comment (miicow) [inline tests/server.test.js:11]: Newline, following airbnb style

**Review by miicow — APPROVED** [review]
Looks good, just some code clean up.

File: jest.config.js (hunk 1 above)
Comment (crystallow1168) [reply jest.config.js:13]: So that jest can handle testing mongoose

**Pull request closed without merging**